### PR TITLE
Add Boyce Index, normalised AUC-PR, and configurable restriction masking

### PR DESCRIPTION
**Pull request by spearmangillman**
- Add boyceIndex() and report Boyce Index + normalised AUC-PR (with cross-validation SDs) in model evaluation output; add prevalence/background info block and random-classifier baseline on AUC-PR plot; fix VariableImportance AUC lookup to use named element
- Add MaskRestrictedAsNoData option (default off) so restriction rasters zero out restricted areas rather than masking to NoData; applies across probability, binary, residuals, MESS, and MoD maps
- Rename update XMLs to drop trailing zeros (e.g. update-2.60.xml → update-2.6.xml); extend update-2.6.xml to add MaskRestrictedAsNoData column migration

### Script Organization

- [ ] Descriptive header (includes name, affiliation, date, brief description of script purpose)
- [ ] Parameters specified at top of script
- [ ] Files loaded at top of script
- [ ] Functions abstracted into separate script
- [ ] Organized into sections

### Readability

- [ ] Repetitive code refactored into functions
- [ ] All functions have docstrings
- [ ] Descriptive in-line comments
- [ ] Correct casing for variables and functions: camelCase (R) or snake_case (Python)
- [ ] Descriptive variable and function names – variables should be nouns and functions should be verbs
- [ ] Used a linter

### Generalizability

- [ ] No hardcoded values (all integers, strings, etc. should be specified at the top of the script in the “Parameters” section)
- [ ] Use relative file paths
- [ ] Consistent output file naming (e.g. “example-filename.tif”)

## Summary by CodeRabbit

* **New Features**
  * Added Boyce Index, prevalence, and normalized AUC-PR metrics to model evaluation results.
  * Added a random-classifier baseline to cross-validation precision-recall plots.
  * Added configurable masking of restricted areas as NoData or zero values.
  * Added configurable fill values for restricted output tiles.
  * Added a warning when the selected binary threshold is below 0.1.
* **Database Updates**
  * Added migrations supporting legacy installations, hyperparameter tuning, Maxent/GAM models, standardized table names, and restricted-area masking configuration.

### CODE REVIEW

File: src/08-fit-model-functions.R [hunk 1]
```diff
@@ -1900,6 +1900,22 @@ makeModelEvalPlots <- function(out = out) {
...
+          linetype = "dotted",
+          color = "black",
+          linewidth = 0.5
+        ) +
+        annotate(
+          "text",
+          x = 0.02, y = prev_line + 0.03,
+          label = paste0("random baseline (prevalence = ", round(prev_line, 3), ")"),
+          hjust = 0, size = 3
+        )
```
[the remark this answers is not in the document]
Comment (spearmangillman) [reply src/08-fit-model-functions.R:1917]: The plot shows both the training PR curve and CV fold curves. Stats$train$prevalence is calculated from Stats$train$auc.data, the same data source as the training curve itself — they're internally consistent. The CV fold curves are subsets of the same training data, so the same baseline applies to all curves on this plot.

**Review by hfadams — APPROVED** [review]
Looks good!
